Dockerfile

```diff
@@ -2,26 +2,33 @@ FROM python:3.9 as builder
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-    build-essential libcairo2-dev libgirepository1.0-dev \
-    gir1.2-ostree-1.0 flatpak
+        build-essential libcairo2-dev libgirepository1.0-dev \
+        gir1.2-ostree-1.0 flatpak
 
-ADD requirements.txt /requirements.txt
-RUN python -m venv /venv && \
-    /venv/bin/pip install -r requirements.txt \
-    && rm -f /requirements.txt
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | /usr/local/bin/python - && \
+    python -m venv /venv && \
+    /venv/bin/python -m pip install -U pip
+
+COPY pyproject.toml poetry.lock /
+
+RUN $HOME/.poetry/bin/poetry export -o requirements.txt && \
+    /venv/bin/pip install -r requirements.txt
 
 FROM python:3.9-slim
+ENV PATH="/venv/bin:$PATH"
 
 EXPOSE 8000
 
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-    libcairo2 gir1.2-ostree-1.0 flatpak && \
-    apt-get clean && rm -rf /var/lib/apt/lists/* && \
-    flatpak --user remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
+        libcairo2 gir1.2-ostree-1.0 flatpak && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+RUN flatpak --user remote-add flathub https://flathub.org/repo/flathub.flatpakrepo && \
+    flatpak --user remote-add flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
 
 COPY ./app /app
 COPY ./data /data
 COPY --from=builder /venv /venv
 
-ENTRYPOINT ["/venv/bin/uvicorn", "--app-dir", "/", "app.main:app", "--host", "0.0.0.0", "--port", "8000", "--reload", "--reload-dir", "/app", "--reload-dir", "/data"]
+ENTRYPOINT ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8000"]
```

Comment:
I don't think we want the reloading to go missing or is this image no longer used for dev?

Reply:
Well, seeing it in context, it might be problematic with this container also being used for smoketests now.

Reply:
`--reload` is still passed to uvicorn in docker-compose: https://github.com/flathub/backend/blob/master/docker-compose.yml#L13, so when you do `docker compose up`:

```
backend_1  | INFO:     Will watch for changes in these directories: ['/app']
backend_1  | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
backend_1  | INFO:     Started reloader process [1] using watchgod
backend_1  | INFO:     Started server process [8]
backend_1  | INFO:     Waiting for application startup.
backend_1  | INFO:     Application startup complete.
```

This shouldn't also cause problems with CPU load as it used `watchgod` now.

Reply:
Somehow missed that change. Should be fine then.